cloudinit/distros/package_management/snap.py

```diff
@@ -35,4 +35,13 @@ def install_packages(self, pkglist: Iterable) -> UninstalledPackages:
 
     @staticmethod
     def upgrade_packages():
-        subp.subp(["snap", "refresh"])
+        (out, _err) = subp.subp(
+            ["snap", "get", "system", "refresh.hold"], rcs=[0, 1]
+        )
+
+        if out.strip() == "forever":
```

Comment:
If I want snap holds enabled, but not by cloud-init, how would I do that with this feature?

If I set `snap refresh --hold=15m`, cloud-init will still attempt the refresh.

I notice that snap returns 1 if this key isn't set. Also, I see what looks like a machine-readable flag `-d`, if parsing is really required.

Reply:
@holmanb this is a good comment, thanks! We should support time-based holds as well as forever. Just added.

Reply:
I had to revert this element because snap daemon itself sets a refresh.hold during early boot, so cloud-init can't distinguish between the snap deamon injected refresh.hold window and something set by user-data.